test/linear_combination.jl

```diff
@@ -161,13 +153,13 @@
         lc = LinearCombination([1, -1])
 
         @testset "all inds" begin
-            @test FeatureTransforms.apply(A, lc) == [-3, -3]
-            @test lc(A) == [-3, -3]
+            @test FeatureTransforms.apply(A, lc; dims=1) == [-3, -3]
+            @test lc(A; dims=1) == [-3, -3]
         end
 
         @testset "dims" begin
             @testset "dims = :" begin
-                @test_throws ArgumentError FeatureTransforms.apply(A, lc; dims=:)
```

Comment:
When we removed the explicit check for `dims=:` the `ArgumentError` is no longer hit.

Now it hits `eachslice(A; dims=:)` which throws a `MethodError` instead .